### PR TITLE
execute tx one by one

### PR DESCRIPTION
**Pull request by Tigo9527**

### CODE REVIEW

File: src/main/java/conflux/dex/service/blockchain/BlockchainSettlementService.java [hunk 1]
```diff
@@ -201,13 +211,51 @@ private void settle(Settleable data) throws Exception {
...
+			waitTxExecuted(data);
+		}
 	}
-	
+
+	private void waitTxExecuted(Settleable data) {
+		String txHash = data.getSettledTxHash();
+		while (true) {
+			// maybe it has been paused by tx monitor already.
+			if (this.healthService.getPauseSource().isPresent()) {
```
Comment (boqiu) [inline src/main/java/conflux/dex/service/blockchain/BlockchainSettlementService.java]: What if pause source is "Manual"?

File: src/main/java/conflux/dex/service/blockchain/BlockchainSettlementService.java [hunk 2]
```diff
@@ -201,13 +211,51 @@ private void settle(Settleable data) throws Exception {
...
+			}
+			Optional<Receipt> opt;
+			try {
+				opt = this.blockchain.getAdmin().getCfx().getTransactionReceipt(txHash).sendAndGet();
+			} catch (RpcException e) {
+				logger.error("rpc exception while fetching tx receipt, tx {}", txHash, e);
+				opt = Optional.empty();
+			}
+			// not executed, or failed
+			if (!opt.isPresent() || opt.get().getOutcomeStatus() != 0) {
```
Comment (boqiu) [inline src/main/java/conflux/dex/service/blockchain/BlockchainSettlementService.java:245]: what if tx failed?

**Review by boqiu — CHANGES_REQUESTED** [review]

File: src/main/java/conflux/dex/service/blockchain/BlockchainSettlementService.java (hunk 1 above)
Comment (Tigo9527) [reply src/main/java/conflux/dex/service/blockchain/BlockchainSettlementService.java]: Changed to `if (this.healthService.isPausedBy(PauseSource.Blockchain))`

File: src/main/java/conflux/dex/service/blockchain/BlockchainSettlementService.java (hunk 2 above)
Comment (Tigo9527) [reply src/main/java/conflux/dex/service/blockchain/BlockchainSettlementService.java:245]: If the transaction fails, the transaction monitor will pause the system, 
 the next loop will break by checking pause source, do nothing here.

**Review by boqiu — APPROVED** [review]

**Pull request merged**